### ISSUE TITLE
namespace param validation doesn't propagate to nested Grape modules

### ISSUE THREAD

**Issue by dantethegrey** [ruby-grape/grape#1085]
Not sure if it is a bug or a feature, but I'd welcome if I could declare parameter validation even for nested API modules like below.

For the former code sample the UserId validator doesn't get called, while for the latter it does.
Behaves the same for builtin validators.
```
params do
  requires :user_id, type: String, user_id: true
end
resource 'user/:user_id' do 
  mount UserProfile
end
```

```
params do
  requires :user_id, type: String, user_id: true
end
resource 'user/:user_id' do 
  get '/' do 
  ...
  end
end
```
Thanks

**Comment by dblock** [comment]
I think it's an interesting corner case, if you can fix it without breaking anything else I would take it.

**Comment by rnubel** [comment]
I believe this fails for basically the same reason as https://github.com/ruby-grape/grape/pull/1050. I'd be curious if [my patch from that issue](https://github.com/intridea/grape/compare/master...rnubel:declared_and_nested_mount_missing_params) also fixes this issue.

**Comment by namusyaka** [comment]
Closed via #1422 
Thanks for reporting the issue, and thanks @rnubel and @plukevdh for your efforts.
Please reopen this issue if you still have a problem.
